src/mpid/ch4/shm/ipc/cma/cma_post.h

```diff
@@ -49,12 +49,12 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_CMA_get_ipc_attr(const void *buf, MPI_Aint co
     if (MPIR_CVAR_CH4_CMA_ENABLE) {
         MPI_Aint data_sz, num_blocks;
         if (HANDLE_IS_BUILTIN(datatype)) {
-            data_sz = MPIR_Datatype_get_basic_size(datatype);
+            data_sz = count * MPIR_Datatype_get_basic_size(datatype);
```

Comment:
Was this path resulting in incorrect data then? Need to backport this patch if so.

Reply:
Yes. I was thinking that we should move `MPIDI_Datatype_check_size` to MPIR layer and establish convention to always use it for getting `data_sz` to prevent such silly mistake.